code/modules/admin/verbs/one_click_antag.dm

```diff
@@ -572,6 +572,7 @@
 		revvie.key = O.key
 		revvie.mind.assigned_role = "revenant"
 		revvie.mind.special_role = "Revenant"
+		revvie.mind.gate_restriction = BOTH_GATES

```

Comment:
Also, you could just put the restriction adding in revenant.dm, around line 270, instead of in two different files.